### PR TITLE
Added atx selection

### PR DESCRIPTION
**Pull request by gavraz**

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -60,6 +60,9 @@ func AddCommands(cmd *cobra.Command) {
 	cmd.PersistentFlags().IntVar(&config.SyncRequestTimeout, "sync-request-timeout",
 		2000, "the timeout in ms for direct requests in the sync")
 
+	cmd.PersistentFlags().IntVar(&config.AtxsSelectCount, "atxs-select-count",
```
Comment (noamnelke) [inline cmd/root.go]: Let's call it `AtxsPerBlock` or something?

File: miner/block_builder.go [hunk 2]
```diff
@@ -29,6 +29,8 @@ const DefaultGas = 1
 
 const IncomingTxProtocol = "TxGossip"
 
+const atxsInBlockLimit = 100
```
Comment (noamnelke) [inline miner/block_builder.go]: This is never used—remove.

**Review by noamnelke — CHANGES_REQUESTED** [review]
Please add a couple of unit tests for this (pool has more/less than limit).

File: miner/block_builder.go (hunk 2 above)
Comment (gavraz) [reply miner/block_builder.go]: Sorry. Used.

File: cmd/root.go (hunk 1 above)
Comment (gavraz) [reply cmd/root.go]: Done

File: miner/block_builder.go (hunk 2 above)
Comment (noamnelke) [reply miner/block_builder.go]: Ok, now I'm just confused... Why do we need a `const` in addition to a config and check that the config is smaller than the const? 🤔

Shouldn't we just trust the config? Alternatively, if we want to validate the config (convince me that we should), we should do it when loading the config, not when using the value.

File: sync/syncer.go [hunk 3]
```diff
@@ -194,28 +196,48 @@ func (s *Syncer) run() {
...
+			return errTooManyAtxs
+		}
+
+		// validate unique tx atx
+		if err := validateUniqueTxAtx(block); err != nil {
+			return err
+		}
+
+		// block eligibility
+		if eligible, err := bv.BlockSignedAndEligible(block); err != nil || !eligible {
```
Comment (barakshani) [inline sync/syncer.go:208]: I would check this first: it is much easier to create a (non-valid) block with unique txs than a block with a valid signature

**Review by noamnelke — APPROVED** [review]

File: sync/syncer.go (hunk 3 above)
Comment (gavraz) [reply sync/syncer.go:208]: Done

**Pull request merged**